tests/ui/generic-const-items/const-trait-impl.rs

```diff
@@ -0,0 +1,23 @@
+// check-pass
+
+// Test that we can call methods from const trait impls inside of generic const items.
+
+#![feature(generic_const_items, const_trait_impl)]
+#![allow(incomplete_features)]
+#![crate_type = "lib"]
+
+const CREATE<T: ~const Create>: T = T::create();
+
+pub const K0: i32 = CREATE::<i32>;
+pub const K1: i32 = CREATE; // arg inferred
+
+#[const_trait]
+trait Create {
+    fn create() -> Self;
+}
+
+impl const Create for i32 {
+    fn create() -> i32 {
+        4096
+    }
+}
```

Comment:
Ooh this is interesting. This should probably happen during astconv when `~const` gets desugared into generic params for the trait. And in this case the generic param should just be `host = false` (disabling the host effect thus making this a const trait bound)

I think this would fit nicely into the PR where constness is removed from `TraitPredicate`.